modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/ColocationGroup.java

```diff
@@ -78,27 +74,20 @@ public static ColocationGroup forAssignments(List<List<UUID>> assignments) {
 
     /** */
     public static ColocationGroup forSourceId(long sourceId) {
-        return new ColocationGroup(Collections.singletonList(sourceId), null, null);
+        return new ColocationGroup(new long[] {sourceId}, null, null);
     }
 
     /** */
     public ColocationGroup() {
     }
 
     /** */
-    private ColocationGroup(List<Long> sourceIds, List<UUID> nodeIds, List<List<UUID>> assignments) {
+    private ColocationGroup(long[] sourceIds, List<UUID> nodeIds, List<List<UUID>> assignments) {
         this.sourceIds = sourceIds;
```

Comment:
possibly it faster than `Arrays.stream(sourceIds).` ?
```suggestion
        this.sourceIds = new GridLongList(sourceIds);
```

Reply:
Stream replaced with `for loop`